libbeat/processors/add_docker_metadata/add_docker_metadata.go

```diff
@@ -123,6 +132,9 @@ func lazyCgroupCacheInit(d *addDockerMetadata) {
 func (d *addDockerMetadata) Run(event *beat.Event) (*beat.Event, error) {
 	var cid string
 	var err error
+	if !d.dockerAvailable {
+		return event, nil
+	}
```

Comment:
Nit. You could move this code to the very end of the function, before these declarations.